misc/v3-task-endpoint-validator/v3-task-endpoint-validator.go

```diff
@@ -412,13 +416,24 @@ func main() {
 		Timeout: 5 * time.Second,
 	}
 
+	// If the image is built with option to check Tags
+	argsWithoutProg := os.Args[1:]
+	if len(argsWithoutProg) > 0 {
+		if argsWithoutProg[0] == "CheckTags" {
+			checkContainerInstanceTags = true
+		}
+	}
+
 	// Wait for the Health information to be ready
 	time.Sleep(5 * time.Second)
 
 	isAWSVPCNetworkMode = false
 	v3BaseEndpoint := os.Getenv(containerMetadataEnvVar)
 	containerMetadataPath := v3BaseEndpoint
 	taskMetadataPath := v3BaseEndpoint + "/task"
+	if checkContainerInstanceTags {
```

Comment:
nit: the logic is right here. However, my preference is using something like
taskMetadataPath := v3BaseEndpoint
if checkContainerInstanceTags {
   taskMetadataPath += "/taskWithTags"
} else {
   taskMetadataPath += "/task"
}
This will make it more clear that what the two different cases are here.

Reply:
Good call. Updated